raiden/network/matrixtransport.py

```diff
@@ -209,8 +210,10 @@ def stop_and_wait(self):
             for async_result in self._messageids_to_asyncresult.values():
                 async_result.set(False)
 
-            gevent.wait(self.greenlets)
-            self._client.logout()
+            try:
+                self._client.join_and_logout(self.greenlets, timeout=self._logout_timeout)
+            except RuntimeWarning as warning:
```

Comment:
Same as in raiden-network/raiden-libs#95, this could just as well be a `RuntimeError`, with `log.error` (or maybe even `log.critical`)